java/java.lsp.server/src/org/netbeans/modules/java/lsp/server/protocol/EqualsHashCodeGenerator.java

```diff
@@ -162,11 +173,10 @@ public CompletableFuture<CodeAction> resolve(NbCodeLanguageClient client, CodeAc
                                 org.netbeans.modules.java.editor.codegen.EqualsHashCodeGenerator.generateEqualsAndHashCode(wc, tp, generateEquals ? selectedFields : null, generateHashCode ? selectedFields : null, -1);
                             }
                         });
-                        if (!edits.isEmpty()) {
-                            codeAction.setEdit(new WorkspaceEdit(Collections.singletonMap(uri, edits)));
-                        }
+                        future.complete(edits.isEmpty() ? null : new WorkspaceEdit(Collections.singletonMap(uri, edits)));
```

Comment:
Would be great if the lsp.api > lsp4j bridge (`Util.workspaceEditFromApi`) worked even here, for `CodeActionProvider`'s commands.

Reply:
The `CodeActionsProvider` implementations from the `java.lsp.server` module do not use the `lsp.api` data structures (`lsp4j` data structures are used directly there). There is no need for using bridge...